rascal-lsp/src/main/rascal/demo/lang/pico/LanguageServer.rsc

```diff
@@ -74,7 +74,7 @@ symbol search in the editor.
 }
 list[DocumentSymbol] picoDocumentSymbolService(start[Program] input)
   = [symbol("<input.src>", DocumentSymbolKind::\file(), input.src, children=[
-      *[symbol("<var.id>", \variable(), var.src) | /IdType var := input]
+      *[symbol("<var.id>", \variable(), var.src) | /IdType var := input, var has id]
```

Comment:
`var has id` is basically the check to see if it's an error node, right? If so, for didactic purposes, it might be instructive to spell it out explicitly, like:

```suggestion
      *[symbol("<var.id>", \variable(), var.src) | /IdType var := input, !hasParseError(var)]
```

With:

```rascal
bool hasParseError(var) = var has id;
```

Or something along those lines. It would be a concrete example to demonstrate how to make a service resilient to recoverable parse errors.

Reply:
We can just use the `util::ErrorRecovery::hasErrors` method. In general this is expensive because the whole tree is traversed, but in this case the tree is small so that is not a problem.